src/domain/entities/DataSet.ts

```diff
@@ -0,0 +1,100 @@
+import { Id, Ref } from "$/domain/entities/Ref";
+import { Maybe } from "$/utils/ts-utils";
+import { BasicDataSet, BasicDataSetAttrs, D2Translation } from "$/domain/entities/BasicDataSet";
+
+export interface DataSetAttrs extends BasicDataSetAttrs {
+    name: string;
+    displayFormName: string;
+    formName: string;
+    sections: Section[];
+    dataSetElements: DataSetElement[];
+}
+
+export class DataSet extends BasicDataSet {
+    name: string;
+    displayFormName: string;
+    formName: string;
+    sections: Section[];
+    dataSetElements: DataSetElement[];
+
+    constructor(attrs: DataSetAttrs) {
+        super(attrs);
+        this.name = attrs.name;
+        this.displayFormName = attrs.displayFormName;
+        this.formName = attrs.formName;
+        this.sections = attrs.sections;
+        this.dataSetElements = attrs.dataSetElements;
+    }
+
+    _getAttributes(): DataSetAttrs {
+        return this._getAttributes() as DataSetAttrs;
+    }
+
+    protected _update(partialAttrs: Partial<DataSetAttrs>): this {
+        Object.assign(this, partialAttrs);
+        return this;
+    }
+
+    static create<DataSet>(
+        this: new (attrs: DataSetAttrs) => DataSet,
+        attrs: DataSetAttrs
+    ): DataSet {
+        return new this(attrs);
+    }
```

Comment:
I think you are having problems with these because you extend "BasicDataSet". Try to import from Struct directly.